api/models/incident.model.js

```
@@ -0,0 +1,152 @@
+    additionalData: mongoose.Schema.Types.Mixed
+  }
+});
+
+// Pre-save middleware to generate incident number
+incidentSchema.pre('save', async function(next) {
+  if (!this.incidentNumber) {
+    try {
+      // Find the highest incident number and increment by 1
+      const highestIncident = await this.constructor.findOne({}, {}, { sort: { 'incidentNumber': -1 } });
+      
+      if (highestIncident && highestIncident.incidentNumber) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)